### PR TITLE
nit(oauth): surface error when token exchange fails in oauth

### PR DESCRIPTION
**Pull request by iamrajjoshi**
image

When something fails in our oauth flow, we have a generic error message, but the error exists in the url. Instead, lets surface it in the html we render.

`pipeline.error` will render the html here:
https://github.com/getsentry/sentry/blob/67c99a3f80384ba767504cebb665f52acb41f616/src/sentry/pipeline/base.py#L208-L212

Now, the html we render will look like
image

### CODE REVIEW

**Review by Christinarlong — APPROVED** [review]
kewllll

**Review by cathteng — APPROVED** [review]
are there tests you need to update?

**Comment by iamrajjoshi** [comment]
> are there tests you need to update?

i checked the tests, they don't assert the template render, just metrics and what the result of the pipeline is 
https://github.com/getsentry/sentry/blob/26c43b35a7e2a48fff22b54001cbc89dd60583fc/tests/sentry/identity/test_oauth2.py#L115-L120

File: src/sentry/identity/oauth2.py [hunk 1]
```diff
@@ -357,7 +357,7 @@ def dispatch(self, request: Request, pipeline) -> HttpResponse:
                 lifecycle.record_failure(
                     "token_exchange_error", extra={"failure_info": ERR_INVALID_STATE}
                 )
-                return pipeline.error(ERR_INVALID_STATE)
+                return pipeline.error(ERR_INVALID_STATE + f"\nError: {error}")
```
Comment (GabeVillalobos) [inline src/sentry/identity/oauth2.py:360]: Very minor nit, but why not just throw the `ERR_INVALID_STATE` string in the fstring string?

**Review by GabeVillalobos — APPROVED** [review]

**Pull request merged**